core-bundle/src/Migration/Version500/AbstractBasicEntitiesMigration.php

```diff
@@ -81,10 +81,12 @@ public function run(): MigrationResult
             );
 
             foreach ($values as $id => $value) {
-                $value = StringUtil::restoreBasicEntities(StringUtil::deserialize($value));
+                $value = StringUtil::deserialize($value);
 
                 if (\is_array($value)) {
-                    $value = serialize($value);
+                    $value = serialize(array_map(static fn ($v) => StringUtil::restoreBasicEntities($v), $value));
```

Comment:
This needs to be done recursive in order to work with even deeper arrays I think.

```suggestion
                    array_walk_recursive($value, static fn (&$v) => $v = StringUtil::restoreBasicEntities($v));
                    $value = serialize($value);
```